vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html

```diff
@@ -991,6 +1181,107 @@
                     });
                 });
                 
+                function getSpaceById(spaceId, callback) {
+                    $.ajax({
+                        url: [[@{|/staff/spaces/search|}]],
+                        dataType: 'json',
+                        success: function(result) {
+                            let spaceName = null;
+                            for (let i = 0; i < result.length; i++) {
+                                const space = result[i];
+                                if (space.id === spaceId) {
+                                    spaceName = space.name;
+                                    break;
+                                }
+                            }
+                            callback(spaceName);
+                        },
+                        error: function(data) {
+                            var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>We are sorry but something went wrong. Please try again later.</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+                            $('.error').append(alert);
+                            $(".error").delay(4000).slideUp(500, function() {
+                                $(".error").empty();
+                            });
+                            // Call the callback with null in case of error
+                            callback(null);
+                        }
+                    });
+                }
+                
+                function getCurrentSpace(blockId) {
```

Comment:
why is this getting `blockId` but is called `getCurrentSpace` and what's the difference to `getSpaceById`?